libass/ass_bitmap.c

```diff
@@ -60,15 +60,14 @@
 
 
 void ass_synth_blur(const BitmapEngine *engine, Bitmap *bm,
-                    int be, double blur_radius)
+                    int be, double blur_r2)
 {
     if (!bm->buffer)
         return;
 
     // Apply gaussian blur
-    double r2 = blur_radius * blur_radius / log(256);
-    if (r2 > 0.001)
-        ass_gaussian_blur(engine, bm, r2);
+    if (blur_r2 > 1e-6)
```

Comment:
I'm not sure that standard guarantee that `exp(0) == 1.0` exactly. Well, that line should distinguish between two discrete values `0.0` and `(TRANSFORM_ACCURACY / 64)^2` so `0.001` is probably fine too.

Reply:
Similar to what I mentioned below, `exp(0) == 1` is indeed guaranteed (as well as `expm1(0) == 0`), but only if the platform declares support for IEC 60559 floating-point arithmetic as per Annex F of the C standard. So I guess we do need to keep some epsilon here. But like you said, `0.001` works equally well here. I don’t really want to say `0.001` is any better than `1e-6`, but this change made me wonder if there was any meaning behind it. Maybe it’s better to keep the old value to avoid unnecessarily wondering the same thing when we dig through this code years later. But maybe I’m just overthinking things.